### ISSUE TITLE
AI Model- Comprehension Quizzes Analyzer

### ISSUE THREAD

**Issue by AprilSurac** [AprilSurac/AI-Model--Book-Summarization-Extraction#2]
iimport gspread
from oauth2client.service_account import ServiceAccountCredentials
import pandas as pd
import tensorflow as tf
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

# Function to authenticate and get worksheet
def authenticate_google_sheet(sheet_name, creds_path):
    scope = ["https://spreadsheets.google.com/feeds", "https://www.googleapis.com/auth/drive"]
    credentials = ServiceAccountCredentials.from_json_keyfile_name(creds_path, scope)
    client = gspread.authorize(credentials)
    sheet = client.open(sheet_name).sheet1  # Assuming the sheet is the first one, adjust as needed
    return sheet

# Function to fetch quiz answers and convert to DataFrame
def fetch_quiz_answers(sheet):
    records = sheet.get_all_records()
    df = pd.DataFrame(records)
    return df

# Function to analyze comprehension quiz results
def analyze_quiz_results(df):
    # Perform analysis based on your requirements
    # Example: Calculate the percentage of correct answers for each question
    for column in df.columns[1:]:
        correct_answers = df[column].value_counts().get("Correct", 0)
        total_responses = df[column].count()
        percentage_correct = (correct_answers / total_responses) * 100
        print(f"Question: {column}, Correct Answers: {correct_answers}/{total_responses}, Percentage Correct: {percentage_correct}%")

    return df  # Return DataFrame for further analysis

# Function to perform logistic regression and display results
def perform_logistic_regression(df):
    # Assuming binary classification with 'Correct' as positive class
    target_column = "Your Target Column Name"  # Replace with your actual target column name
    features = df.drop(columns=[target_column])
    labels = (df[target_column] == "Correct").astype(int)

    # Split data into training and testing sets
    X_train, X_test, y_train, y_test = train_test_split(features, labels, test_size=0.2, random_state=42)

    # Logistic Regression Model
    model = LogisticRegression()
    model.fit(X_train, y_train)

    # Evaluate the model
    y_pred = model.predict(X_test)
    accuracy = accuracy_score(y_test, y_pred)
    print(f"Logistic Regression Accuracy: {accuracy * 100:.2f}%")

    # Display graphical output
    plt.scatter(X_test, y_test, color='black', label='Actual')
    plt.scatter(X_test, y_pred, color='blue', label='Predicted', marker='x')
    plt.xlabel("Feature")
    plt.ylabel("Correct/Incorrect (1/0)")
    plt.legend()
    plt.show()

# Example Usage
spreadsheet_name = "Your Spreadsheet Name"  # Replace with your spreadsheet name
credentials_path = "path/to/your/credentials.json"  # Replace with your credentials JSON file path
quiz_sheet = authenticate_google_sheet(spreadsheet_name, credentials_path)
quiz_df = fetch_quiz_answers(quiz_sheet)
analyzed_df = analyze_quiz_results(quiz_df)
perform_logistic_regression(analyzed_df)
